osbrain/tests/test_agent.py

```diff
@@ -39,6 +39,17 @@ def test_agent_uuid():
     assert all(isinstance(identifier, bytes) for identifier in bunch)
 
 
+def test_agent_auto_generated_name(nsproxy):
+    """
+    If an agent is initialized without a name, a globally unique one must be
+    auto generated for it.
+    """
+    agent = run_agent()
+    name = agent.get_attr('name')
+    assert isinstance(name, str)
+    assert name != ''
+
```

Comment:
Yeah, this is a minimal check. But I would rather do something like:

```python
def test_xxx(nsproxy):
    N = 10
    names = [run_agent().get_attr('name') for x in range(N)]
    assert all(isinstance(name, str) for name in names)
    assert len(set(names)) == N
    assert set(names) == set(nsproxy.agents())
```

Reply:
That's one way to check they don't collide. Still not perfect, though, but I can't think of anything better.